importer/downloader.py

```diff
@@ -5,19 +5,41 @@
 import sys
 from six.moves import urllib
 
-searchurl = "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&term=Retracted+Publication&retmax=10&retmode=json"
+searchurl = "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&term=Retracted+Publication&retmax=6000&retmode=json"
 fetchurl = "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?&db=pubmed&rettype=xml&id="
 request = urllib.request.Request(searchurl)
 
+#Attempt to download the given url, returning the xml document
+#Exits after 5 failed attempts
+def attemptDownload(downloadurl,retries=0):
+    if retries > 5:
+        print("download failed")
+        sys.exit
+    try:
+        with contextlib.closing(urllib.request.urlopen(downloadurl)) as response:
+            data = response.read()
+            return data
+    except:
+        retries+=1
+        attemptDownload(downloadurl,retries)
+
+def attemptIdList(retries=0):
+    if retries > 5:
+        print("failed to retrieve ID list")
```

Comment:
Use `sys.stderr.write` here instead of `print` (or at least `print("..", file=sys.stderr)` so that we don't end up piping the output of this error to a JSON parser.